src/auth/services/auth.service.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { SingInAuthDto } from '../dto/create-auth.dto';
+import { AuthRepository } from '../repository/auth-repository';
+import { User } from '../entities/user.entity';
+import * as bcrypt from 'bcrypt';
+import { LoginAuthDto } from '../dto/login-auth.dto';
+import { JwtTokenService } from './jwt.service';
+import { JwtRepository } from '../repository/jwt-repository';
+import { JwtPayload } from 'src/common/strategies/accessToken.strategy';
+import { EmojiLogger } from 'src/common/logger/EmojiLogger';
+import { verifyCode } from 'src/utils/verify/verifyCode';
+import { VerificationRepository } from '../repository/verification-repository';
+import { SendCodeService } from './sendCode.service';
+
+@Injectable()
+export class AuthService {
+  private readonly logger = new EmojiLogger();
+  constructor(
+    private readonly jwtTokenService: JwtTokenService,
+    private readonly sendCodeService: SendCodeService,
+    private readonly authRepository: AuthRepository,
+    private readonly jwtRepository: JwtRepository,
+    private readonly verificationRepository: VerificationRepository,
+  ) {}
+
+  async create(singInAuthDto: SingInAuthDto): Promise<{ data: User }> {
+    singInAuthDto.password = await bcrypt
+      .hash(singInAuthDto.password, 10)
+      .catch((error) => {
+        this.logger.error(error);
+        throw new InternalServerErrorException('Server Error');
+      });
+
+    delete singInAuthDto.passwordRepeat;
+
+    const verificationCode = await verifyCode();
+
+    const user: User = await this.authRepository
+      .create(singInAuthDto)
+      .catch((error) => {
+        this.logger.error(`create user: ${error}`);
+        throw new InternalServerErrorException('Server Error');
+      });
+    delete user.dataValues.password;
+
+    await this.verificationRepository
```

Comment:
Your `create` method does to much. It creates user, creates the verification code, invoke the `sendCodeService.send` method. 

In general it works and it can be OK. But we're studying and because of it I want to suggest to you to consider the next approaches.

First - decompose this method and move the sequence in the controller. E.g.

```ts
//.....
const user = await authService.create(signInAuthDTO);
const verificationCode = await verificationService.create(user.id);
await this.sendCodeService.send({...signInAuthDTO, code: verificationCode});

return Response.success({data: user});
```

In this case services will be granular and not coupled, and controller method will be as a director who only command to each service.

Second - read about CQRS pattern https://docs.nestjs.com/recipes/cqrs I don't push you to implement it because it's quite complex thing, but I want that you know it.